src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +34,260 @@ public void GetSerializableMembers_Interface()
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_NetCore_TestData))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_NetCore_ThrowsNotSupportedException(Type type)
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The coreclr doesn't support GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_Netfx_ThrowsNotSupportedException(Type type)
+        {
+            Assert.NotNull(FormatterServices.GetUninitializedObject(type));
+            Assert.NotNull(FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for shared generic instances")]
+        public void GetUniniitalizedObject_SharedGenericInstance_NetCore_ThrowsNotSupportedException()
+        {
+            Type canonType = Type.GetType("System.__Canon");
```

Comment:
I do not think it is worth the pain to test the behavior for internal CoreLib types. The behavior for internal CoreLib types will differ from runtime to runtime. These tests would need to have yet another customization for uataop or for Mono.